tools/logger/filter.c

```diff
@@ -0,0 +1,172 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ * Author: Karol Trzcinski	<karolx.trzcinski@linux.intel.com>
+ */
+
+#include <smex/ldc.h>
+#include <sof/lib/uuid.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "convert.h"
+#include "filter.h"
+
+int parse_uuid(const char *src, struct sof_uuid *dst)
+{
+	uint32_t b;
+	uint32_t c;
+	uint32_t d0;
+	uint64_t d2;
+	int n;
+	int ret;
+
+	ret = sscanf(src, " <%X-%X-%X-%X-%lX>%n", &dst->a, &b, &c, &d0, &d2,
+		     &n);
+	if (ret != 5 || b > UINT32_MAX || c > UINT32_MAX)
+		return -EINVAL;
```

Comment:
I suppose clobbering dst->a is fine but clobbering dst->b or dst->c is not?

Also I seriously doubt you can ever get b > UINT32_MAX given that b is a uint32_t.